react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { clickCount: 0 };
+    this.count = 0;
```

Comment:
With a few exceptions, we want to avoid attaching any additional information to the `this` object. The information can/should almost always be stored in state.